modules/host.go

```diff
@@ -83,49 +93,113 @@ type (
 		UploadCalls       uint64 `json:"uploadcalls"`
 	}
 
+	// StorageFolderMetadata contians metadata about a storage folder that is
+	// tracked by the storage folder manager.
+	StorageFolderMetadata struct {
+		CapacityRemaining uint64
+		Path              string
+		TotalCapacity     uint64
+
+		// Successful and unsuccessful operations report the number of
+		// successful and unsuccessful disk operations that the storage manager
+		// has completed on the storage folder. A large number of unsuccessful
+		// operations can indicate that the space allocated for the storage
+		// folder is larger than the amount of actual free space on the disk.
+		// Things like filesystem overhead can reduce the amount of actual
+		// storage available on disk, but should ultimately be less than 1% of
+		// the total advertised capacity of a disk. A large number of
+		// unsuccessful operations can also indicate that the disk is failing
+		// and that it needs to be replaced.
+		SuccessfulOperations   uint64
+		UnsuccessfulOperations uint64
+	}
+
+	// StorageFolderManager tracks and manipulates storage folders. Storage
+	// folders are used by the host to store the data that they contractually
+	// agree to manage.
+	StorageFolderManager interface {
+		// AddStorageFolder adds a storage folder to the host. The host may not
+		// check that there is enough space available on-disk to support as
+		// much storage as requested, though the host should gracefully handle
+		// running out of storage unexpectedly.
+		AddStorageFolder(path string, size uint64) error
+
+		// ForceRemoveStorageFolder removes a storage folder. The host will try
+		// to save the data on the storage folder by moving it to another
+		// storage folder, but if there are errors (such as not enough space)
+		// the host will forcibly remove the storage folder, discarding the
+		// data. This means that the host will be unable to provide storage
+		// proofs on the data, and is going to incur penalties.
+		ForceRemoveStorageFolder(index int) error
+
+		// RemoveStorageFolder will remove a storage folder from the host. All
+		// storage on the folder will be moved to other storage folders,
+		// meaning that no data will be lost. If the host is unable to save
+		// data, an error will be returned and the operation will be stopped.
+		RemoveStorageFolder(index int) error
+
+		// ResetStorageFolderHealth will reset the health statistics on a
+		// storage folder.
+		ResetStorageFolderHealth(index int) error
```

Comment:
because you'd get two different UIDs when adding the same folder two different times? Hmm.
From an outsider perspective, what sticks out to me is that the index doesn't seem to carry any meaning. It implies that there's an ordering, but it's not obvious what that ordering is or why it would be useful. With a UID, it's clear that there's no hidden meaning; it's just a token used to reference a resource.
A UID would also let you use a map (of `*storageFolder`s) internally instead of a slice, which seems like it might be preferable, though I can't say for sure.